.github/workflows/bytemotion-node-ci.yml

```diff
@@ -43,16 +43,32 @@ jobs:
         with:
           node-version: 14
           registry-url: https://registry.npmjs.org/
-      - run: | 
-          git config --global user.email "${{secrets.GIT_USER_EMAIL}}"
-          git config --global user.name "${{secrets.GIT_USER_NAME}}"
+
+      - name: Set Minor Release
+        id: minor_release
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.merged == true }}
+        run: echo "::set-output name=RELEASE::minor"
+
+      - name: Set PreRelease Release
+        id: prerelease_release
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.merged == false }}
+        run: echo "::set-output name=RELEASE::prerelease --preid=rc"
+
+      - name: Tag and Publish
+        run: | 
           BRANCH=$(git rev-parse --abbrev-ref HEAD)
-          [[ $BRANCH = 'master' ]] && RELEASE="patch" || RELEASE="prerelease --preid=rc"
+          RELEASE=${{ 
+            steps.minor_release.outputs.RELEASE ||
+            steps.prerelease_release.outputs.RELEASE ||
+            'patch'
+          }}
+          git config --global user.email "${{ secrets.GIT_USER_EMAIL }}"
+          git config --global user.name "${{ secrets.GIT_USER_NAME }}"
           npm install
           npm run build
           git checkout -- .
           npm version $RELEASE
-          echo "Pushing to $BRANCH"
+          echo "Pushing $RELEASE to $BRANCH"
           git push origin $BRANCH
           git push origin $BRANCH --tags
           npm publish
```

Comment:
What I'm lacking is to have option with changing "major" version. But I don't get whole concept behind this and... I suppose this should be done manually in `package.json` right? 
BTW some time ago I seen option which uses keywords in commit message to manual steering of these version numbers things...

Reply:
Yeah, i was thinking about that too. Major seems like something that should be done manually? Yeah, maybe the key word is a good option. Major versions are rare though and some of these libraries might never actually have another major release :smile:

Reply:
So I think the best way to determine a major release is when creating one manually through githubs formal release mechanism. I've added it, what do you thin @wooyey

Reply:
Seems legit to me. Now only some place to document this briefly, at least for newcomers... and myself for some time ;)

Reply:
nice! I added a readme with descriptions of the conditions necessary for each release type